### PR TITLE
Team mapml refactoring

### PR DESCRIPTION
**Pull request by prushforth**
Fixes issue #511

### CODE REVIEW

File: src/mapml/layers/MapLayer.js [hunk 1]
```diff
@@ -1120,15 +1146,15 @@ export var MapMLLayer = L.Layer.extend({
       let extent = this._extent;
       if(!extent) return FALLBACK_PROJECTION;
       switch (extent.tagName.toUpperCase()) {
-        case "EXTENT":
+        case "MAP-EXTENT":
           if(extent.hasAttribute('units'))
             return extent.getAttribute('units').toUpperCase();
           break;
         case "INPUT":
```
Comment (ahmadayubi) [inline src/mapml/layers/MapLayer.js:1153]: I believe this has to be updated as well since the test files use map-input rather than input now.

Comment (ahmadayubi) [inline src/mapml/layers/MapLayer.js:1153]: I believe this has to be updated as well since the test files use map-input rather than input now.

**Comment by prushforth** [comment]
@Malvoz @erictheise fyi this is a big change, hopefully we've covered it all.  Let us know if you have concerns.  We'll push the change on geogratis and experiments when this gets merged.

**Comment by ahmadayubi** [comment]
Should `<layer->` be changed to `<map-layer>`?

**Comment by prushforth** [comment]
> Should `<layer->` be changed to `<map-layer>`?

I believe it would be quite disruptive, and the gain would be a bit more coherence. I tend towards "no, not enough benefit", but remain open to opinions.

**Comment by Malvoz** [comment]
If we want to be consistent we'll also have to change `<mapml->` to `<map-mapml>` which reads a bit funny. I personally squinted at `<layer->` the first time I saw it, while it is valid HTML it looks a bit awkward before getting accustomed to. We should consider if it may affect how developers, especially beginners, perceive them. And if we're to make the changes, they should probably happen in this PR.

I don't have a strong opinion on this, I'm mostly happy the polyfill is heading towards web platform compatibility. 😊

**Comment by ahmadayubi** [comment]
Line 965, it's creating input elements for an extent, I believe they should be `<map-input>` rather than `<input>`. Although the fact that it didn't cause errors may mean it's either not a function being used anymore or it's not tested for.

**Comment by prushforth** [comment]
> Although the fact that it didn't cause errors may mean it's either not a function being used anymore or it's not tested for.

I think it's this, but I'll remove that code once we've got this issue closed.  Should have done it before but tempes fugit :-)

File: test/e2e/data/cbmt.mapml [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<mapml- xmlns="http://www.w3.org/1999/xhtml/">
+  <map-head>
+    <map-title>Canada Base Map - Transportation (CBMT)</map-title>
+    <map-meta http-equiv="Content-Type" content="text/mapml;projection=OSMTILE"></map-meta>
+    <map-meta charset="utf-8"></map-meta>
+    <map-link rel="license" href="https://www.nrcan.gc.ca/earth-sciences/geography/topographic-information/free-data-geogratis/licence/17285" map-title="Canada Base Map © Natural Resources Canada"></map-link>
```
Comment (Malvoz) [inline test/e2e/data/cbmt.mapml:6]: @prushforth I'm looking to reflect these changes in https://github.com/Maps4HTML/MapServer-documentation/tree/mapml, I'm wondering whether it was intended to change the `title` attribute on `<map-link>`(s) to `map-title` as well.

Comment (Malvoz) [reply test/e2e/data/cbmt.mapml:6]: I'm assuming that was not intended, as it seems no other attributes use the `map-` prefix.

Comment (prushforth) [reply test/e2e/data/cbmt.mapml:6]: That's correct. We only changed the elements' prefix, I'll await further information as to whether it's necessary to change attributes. Fwiw I think the link title attribute would be appropriate if there's a conflict there that might be motivation to adopt map-title. We haven't done that yet, so I think it's best to keep with title for now.  Btw I can't find the mapserver MapML code, I was looking to update it too. Lete know if you come across it.

Comment (Malvoz) [reply test/e2e/data/cbmt.mapml:6]: MapServer doesn't actually support MapML yet, here's the implementation ticket: https://github.com/MapServer/MapServer/issues/6024.

**Pull request merged**